tests/basic/upgrade_test.go

```diff
@@ -504,3 +505,198 @@ var _ = Describe("{UpgradeVolumeDriverFromCatalog}", func() {
 		AfterEachTest(contexts)
 	})
 })
+
+var _ = Describe("{UpgradePxKvdbMemberDown}", func() {
+	/*
+					https://purestorage.atlassian.net/browse/PTX-21450
+				    https://portworx.testrail.net/index.php?/cases/view/94371
+					1. Bring down a kvdb member node
+			        2. Upgrade PX
+			        3. Bring up the kvdb member node
+		            4. Verify if the kvdb member joins the cluster and everything is running fine
+	*/
+
+	StartTorpedoTest("UpgradePxKvdbMemberDown", "Validating PX KVDB member down during upgrade", nil, 0)
+
+	var contexts []*scheduler.Context
+
+	var timeBeforeUpgrade time.Time
+	var timeAfterUpgrade time.Time
+	var kvdbNode node.Node
+
+	itLog := "Upgrade PX when kvdb member is down and after upgrade the kvdb member should join the kvdb cluster"
+	It(itLog, func() {
+		log.InfoD(itLog)
+		storageNodes, err := GetStorageNodes()
+		log.FailOnError(err, "Failed to get storage nodes")
+		numOfNodes := len(storageNodes)
+
+		stepLog := "schedule applications"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			for i := 0; i < Inst().GlobalScaleFactor; i++ {
+				contexts = append(contexts, ScheduleApplications(fmt.Sprintf("upgradevolumedriver-%d", i))...)
+			}
+		})
+		ValidateApplications(contexts)
+		defer DestroyApps(contexts, nil)
+
+		if len(Inst().UpgradeStorageDriverEndpointList) == 0 {
+			log.Fatalf("Unable to perform volume driver upgrade hops, none were given")
+		}
+
+		for _, upgradeHop := range strings.Split(Inst().UpgradeStorageDriverEndpointList, ",") {
+			stepLog = "Add labels to non kvdb nodes so that they don't try to become kvdb node"
+			Step(stepLog, func() {
+				log.InfoD(stepLog)
+				for _, n := range storageNodes {
+					kvdbFlag, err := CheckIfNodeIsKvdbNode(n)
+					log.FailOnError(err, "Failed to check if node is kvdb node")
+					if !kvdbFlag {
+						err = Inst().S.AddLabelOnNode(n, k8s.MetaDataLabel, "false")
```

Comment:
In this step we are labeling non kvdb members so that when the kvdb node goes down it doesn't immediately become kvdb member this is according to the bug itself.